### PR TITLE
LP-1950 Create Idea and Marketplace App and add all possible pages and their URLs

### PR DESCRIPTION
**Pull request by qasimgulzar**
### Description

[LP-1950](https://philanthropyu.atlassian.net/browse/LP-1950)

[philu-edx-theme PR](https://github.com/philanthropy-u/philu-edx-theme/pull/1155)

### CODE REVIEW

**Comment by zeshanarifios** [comment]
@emraan004
Merge pull request #1446 from philanthropy-u/develop …
011b2ba
AhmedFarazAhmedFaraz
Change dependencies to merge regflow
811c420
@emraan004
Merge pull request #1447 from philanthropy-u/faraz/resolve_onboarding… …
ecf720d
@ahmed-faraz46
Merge pull request #1450 from philanthropy-u/develop …
0fe5251
@emraan004
Merge pull request #1453 from philanthropy-u/usman/LP-1709 …
8fac851

Please have a look why these commits are part of your PR?

**Comment by qasimgulzar** [comment]
> Why are we creating these apps in a newly created directory `philu/djangoapps` instead of putting these in `openedx/features`?

@mabdullahz these apps have nothing to do with openedx thats why we are putting out of it.

**Comment by mabdullahz** [comment]
> > Why are we creating these apps in a newly created directory `philu/djangoapps` instead of putting these in `openedx/features`?
> 
> @mabdullahz these apps have nothing to do with openedx thats why we are putting out of it.

Was this discussed in a meeting I was not present in? This sounds like something we should disucss internally for sure.

**Comment by qasimgulzar** [comment]
> > > Why are we creating these apps in a newly created directory `philu/djangoapps` instead of putting these in `openedx/features`?
> > 
> > 
> > @mabdullahz these apps have nothing to do with openedx thats why we are putting out of it.
> 
> Was this discussed in a meeting I was not present in? This sounds like something we should disucss internally for sure.

yes, we had discussed it yesterday's refinement meeting.

**Comment by mabdullahz** [comment]
> > > > Why are we creating these apps in a newly created directory `philu/djangoapps` instead of putting these in `openedx/features`?
> > > 
> > > 
> > > @mabdullahz these apps have nothing to do with openedx thats why we are putting out of it.
> > 
> > 
> > Was this discussed in a meeting I was not present in? This sounds like something we should disucss internally for sure.
> 
> yes, we had discussed it yesterday's refinement meeting.

So are we also going to be moving the custom apps that are currently in `openedx/features` out to `philu/djangoapps` now? If not, this is going to result in a disorder, where:

1. Others are not sure where to put the next app.
2. Others are not sure where to find the existing apps.

**Comment by qasimgulzar** [comment]
> > > > > Why are we creating these apps in a newly created directory `philu/djangoapps` instead of putting these in `openedx/features`?
> > > > 
> > > > 
> > > > @mabdullahz these apps have nothing to do with openedx thats why we are putting out of it.
> > > 
> > > 
> > > Was this discussed in a meeting I was not present in? This sounds like something we should disucss internally for sure.
> > 
> > 
> > yes, we had discussed it yesterday's refinement meeting.
> 
> So are we also going to be moving the custom apps that are currently in `openedx/features` out to `philu/djangoapps` now? If not, this is going to result in a disorder, where:
> 
> 1. Others are not sure where to put the next app.
> 2. Others are not sure where to find the existing apps.

if those are not related to independent functionalities from EdX, for sure we should move them as well.

**Comment by zeshanarifios** [comment]
@qasimgulzar 
As discussed in meething, Please move apps from philu package to openedx/features. and change the theme side accordingly. 
Thank you

File: lms/envs/common.py [hunk 1]
```diff
@@ -2397,7 +2397,10 @@ def _make_locale_paths(settings):
     # User Leads app
     'openedx.features.user_leads',
 
-    'channels'
+    'channels',
+
+    'openedx.features.marketplace',
```
Comment (ahmed-faraz46) [inline lms/envs/common.py:2402]: Let's please move these under `# Philu Features` comment at line 2368 in the same file.

File: openedx/features/idea/urls.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from django.conf.urls import url
+
+from openedx.features.idea.views import (ChallengeLandingView, IdeaListingView, IdeaCreateView, IdeaDetailView)
+
+urlpatterns = [
+    url(
+        r'^ideas/overview$',
```
Comment (ahmed-faraz46) [inline openedx/features/idea/urls.py:7]: How about if move `^idea/` in lms.urls where we plugged these urls?

File: openedx/features/idea/views.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+# Create your views here.
```
Comment (ahmed-faraz46) [inline openedx/features/idea/views.py:4]: Let's please remove this comment as it's not serving any purpose. OR correct me if I am wrong?

File: openedx/features/idea/views.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
```
Comment (ahmed-faraz46) [inline openedx/features/idea/views.py:1]: Is necessary to keep this comment? if not let's please remove it from all files.

File: openedx/features/marketplace/urls.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
+from django.conf.urls import url
+
+from openedx.features.marketplace.views import (MarketplaceListingView, MarketplaceCreateRequestView,
+                                                MarketplaceRequestDetailView)
+
+urlpatterns = [
+    url(
+        r'^marketplaces/$',
```
Comment (ahmed-faraz46) [inline openedx/features/marketplace/urls.py:8]: How about if move `^marketplaces/` in lms.urls where we plugged these urls?

File: openedx/features/marketplace/views.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+# Create your views here.
```
Comment (ahmed-faraz46) [inline openedx/features/marketplace/views.py:4]: Let's please remove this comment as it's not serving any purpose. OR correct me if I am wrong?

File: openedx/features/idea/apps.py [hunk 7]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
```
Comment (ahmed-faraz46) [inline openedx/features/idea/apps.py:2]: I see this import in every file, we aren't using it anywhere, is it necessary to keep?

File: lms/urls.py [hunk 8]
```diff
@@ -1092,6 +1092,8 @@
 # edx-drf-extensions csrf app
 urlpatterns += [
     url(r'', include('csrf.urls')),
+    url(r'', include('openedx.features.marketplace.urls')),
```
Comment (ahmed-faraz46) [inline lms/urls.py:1095]: Let's please move at line `91` under comment `#Philu features`

Comment (ahmed-faraz46) [inline lms/urls.py:1095]: Let's please add `^idea/` here.

File: lms/urls.py [hunk 9]
```diff
@@ -1092,6 +1092,8 @@
...
+    url(r'', include('openedx.features.idea.urls')),
```
Comment (ahmed-faraz46) [inline lms/urls.py:1096]: Let's please add `^marketplace` here.

File: openedx/features/idea/apps.py (hunk 7 above)
Comment (qasimgulzar) [reply openedx/features/idea/apps.py:2]: https://www.quora.com/What-is-unicode_literals-in-Python

File: openedx/features/idea/views.py (hunk 4 above)
Comment (qasimgulzar) [reply openedx/features/idea/views.py:1]: it is good to have to inform the developer about the encoding as we a `unicode_literals `

**Pull request merged**